.github/workflows/ci.yml

```diff
@@ -0,0 +1,57 @@
+on: [push]
+
+jobs:
+  build_and_test:
+    name: Build adn test on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+
+    steps:
+    - uses: actions/checkout@master
+
+    - name: Install nightly
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: nightly
+        override: true
+
+    - name: check
+      uses: actions-rs/cargo@v1
+      with:
+        command: check
+        args: --all --benches --bins --examples --tests
+
+    - name: check unstable
+      uses: actions-rs/cargo@v1
+      with:
+        command:  check
+        args: --features unstable --all --benches --bins --examples --tests
+
+    - name: tests
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --all --doc --features unstable
+
+  check_fmt_and_docs:
+    name: Checking fmt and docs
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+
+    - name: setup
+      run: |
+        rustup default nightly
+        rustup component add rustfmt
+        cargo install mdbook
+        rustc --version
```

Comment:
Did you see the [install-mdbook](https://github.com/async-rs/async-std/blob/master/ci/install-mdbook.sh) script? That way saves quite a bit of time compared to the `cargo install mdbook`.

However, the script is fragile because it ultimately relies on a specific release filename pattern and it would be nice if we could have a more general solution. (This is currently an issue for another CI related PR [here](https://github.com/async-rs/async-std/pull/138)).

I think a better way might be to create a docker image based on the official image (or slim or docs.rs) with all of the tools we need (deadlinks, mdbook, tarpaulin?) pre-installed. Then we could use the [container.image setting](https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idcontainerimage) to set the environment so that stuff is available without having to rebuild it.

Another github action could handle docker image related tasks.

Reply:
I agree this would be really nice, and it would probably worth doing this in a follow-up!

Perhaps it's worth opening an issue for this on https://github.com/actions-rs/meta/issues also; I agree that having a general solution for this would probably be very valuable to a lot of people.

Reply:
`cargo install mdbook` should be avoided, it adds ~8 min of build time.

Reply:
Wow, long.  I will fix it tonight, so take a moment